### PR TITLE
Fix example in README

### LINKED ISSUE #52: `README` example does not run; opens in read-only by default
In the `iohub`'s README we have:

```
with open_ome_zarr("20200812-CardiomyocyteDifferentiation14-Cycle1.zarr") as dataset:
    ...
    new_fov = dataset.create_position("A", "1", "0")  # creates a new fov
    ...
```
which fails because by default `open_ome_zarr` opens in read-only mode. 

Adding `mode='r+'` works, but it modified/added to the dataset I downloaded from the web, then failed on subsequent runs (since the position already existed when I ran it the first time). 

Maybe a more user-friendly README example should read the example dataset, write to a new file, then read+append to that new file?

### PR DESCRIPTION
**Pull request by ziw-liu**
Fix #52

### CODE REVIEW

**Comment by ziw-liu** [comment]
@talonchandler Note that writing to a new store results in a longer example section.

File: README.md [hunk 1]
```diff
@@ -34,13 +34,25 @@ Load and modify an [official example OME-Zarr](https://zenodo.org/record/7274533
...
+    raw_data = img_array.numpy()  # loads a CZYX 4D array into RAM
+    print(raw_data.mean())  # does some analysis
+
+with open_ome_zarr(
+    "max_intensity_projection.zarr",
+    mode="w-",
+    layout="hcs",
+    channel_names=channel_names,
+) as dataset:
+    dataset.axes = dataset._DEFAULT_AXES[-2:]  # reduce TCZYX axes to only XY
```
Comment (talonchandler) [inline README.md:53]: Can we skip this and let TCZ have axes lengths 1?

The underscore in `_DEFAULT_AXES` tells me "don't use this" which is a bit inconsistent with showing this in the README example IMO.

File: README.md [hunk 2]
```diff
@@ -34,13 +34,25 @@ Load and modify an [official example OME-Zarr](https://zenodo.org/record/7274533
...
-    first_fov = dataset["B/03/0"]  # lazy Zarr group
-    data = first_fov["0"].numpy()  # loads a CZYX 4D array into RAM
-    print(data.mean())  # does some analysis
-    new_fov = dataset.create_position("A", "1", "0")  # creates a new fov
-    new_fov["0"] = np.ones(data.shape)  # writes some ones to a new Zarr array
+    channel_names = dataset.channel_names
+    print(channel_names)
+    img_array = dataset[
+        "B/03/0/0"
+    ]  # lazy Zarr array for the raw image in the first position
```
Comment (talonchandler) [inline README.md:43]: Just formatting, but does this need to be split across multiple lines?

File: README.md [hunk 3]
```diff
@@ -34,13 +34,25 @@ Load and modify an [official example OME-Zarr](https://zenodo.org/record/7274533
 import numpy as np
 from iohub.ngff import open_ome_zarr
 
-with open_ome_zarr("20200812-CardiomyocyteDifferentiation14-Cycle1.zarr") as dataset:
+with open_ome_zarr("test.zarr", mode="r", layout="auto") as dataset:
```
Comment (talonchandler) [inline README.md:37]: I actually don't mind the previous idea of having the specific dataset name along with the link directly above it.

Comment (talonchandler) [reply README.md:37]: I'm open to other solutions, though. Maybe there's a faster way of getting a small `test.zarr` dataset in the user's working directory.

File: README.md [hunk 4]
```diff
@@ -34,13 +34,25 @@ Load and modify an [official example OME-Zarr](https://zenodo.org/record/7274533
...
+    ]  # lazy Zarr array for the raw image in the first position
+    raw_data = img_array.numpy()  # loads a CZYX 4D array into RAM
+    print(raw_data.mean())  # does some analysis
+
+with open_ome_zarr(
+    "max_intensity_projection.zarr",
+    mode="w-",
+    layout="hcs",
+    channel_names=channel_names,
+) as dataset:
```
Comment (talonchandler) [inline README.md:52]: Again, just formatting, but it's kind of annoying that it's formatted one way here and another way above.

File: README.md (hunk 3 above)
Comment (ziw-liu) [reply README.md:37]: Yes the original file name is better. Reverting

File: README.md (hunk 4 above)
Comment (ziw-liu) [reply README.md:52]: This is black behavior. So I guess it's 'consistent' in its own way? I'm open to manually match formatting here though.

File: README.md (hunk 1 above)
Comment (ziw-liu) [reply README.md:53]: Good point!

File: README.md (hunk 4 above)
Comment (ziw-liu) [reply README.md:57]: Not sure if this helps but now the first one is also line-wrapped.

**Review by talonchandler — APPROVED** [review]
Looks good! I'm flexible on formatting.

**Pull request merged**